src/cript/api/paginator.py

```diff
@@ -193,11 +193,17 @@ def fetch_page_from_api(self) -> List[dict]:
         ).json()
 
         # handling both cases in case there is result inside of data or just data
-        if "result" in response["data"]:
+        try:
             self.current_page_results = response["data"]["result"]
-        else:
+        except KeyError:
+            self.current_page_results = response["data"]
+        except TypeError:
             self.current_page_results = response["data"]
 
+        if response["code"] == 404 and response["error"] == "The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.":
```

Comment:
do you want to put the paginator refactoring in a different PR?

Reply:
has this been tested with the search test functions?

Reply:
Well, the paginator was raising a cryptic exception when searching for a non-existent node.

But for the patch solution, we need it to return an empty list in that case.

I put it in a different commit, but I need this here to work.

Reply:
got it!

We have to come back to paginator later and really beef it up and refactor it because when I first wrote it it was just a rough draft of what I knew then